src/AspNet.Security.OAuth.Instagram/InstagramAuthenticationHandler.cs

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Extensions;
+using Microsoft.AspNet.Authentication;
+using Microsoft.AspNet.Authentication.OAuth;
+using Microsoft.AspNet.Http.Authentication;
+using Microsoft.Extensions.Internal;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Instagram {
+    public class InstagramAuthenticationHandler : OAuthHandler<InstagramAuthenticationOptions> {
+        public InstagramAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {}
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var userInformationEndpoint = $"{Options.UserInformationEndpoint}?access_token={tokens.AccessToken}";
+
+            if (Options.SignedRequestsEnforced) {
```

Comment:
Not to mention that this option will only impact the userinfo request, which is made before the access token is persisted some (it's still in memory at this point), so the security gain is really marginal.